src/livekit/openIDSFU.ts

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient } from "matrix-js-sdk";
+import { logger } from "matrix-js-sdk/src/logger";
+
+import { Config } from "../config/Config";
+
+export interface SFUConfig {
+  url: string;
+  jwt: string;
+}
+
+export async function getSFUConfigWithOpenID(
+  client: MatrixClient,
```

Comment:
It feels like adding `MatrixClient` is a quite large dependency for this function that we probably don't want to have as it adds too much power/responsibility for this function to handle 🙂 From what I can see, we only require 2 fields from the `client`. It would be safer if we could pass these fields directly.

Perhaps instead of `client`, we could pass a single small structure, something like `clientIdentity`, or, in case we only need a device ID, just a device ID would do.

Reply:
Yeah, seems fair enough, I was mostly just copying the way it was done before. Done.

Reply:
Thanks! I think that we actually probably don't even need to use `Pick<>`, because it mentions `MatrixClient` (as if we're saying that we need only parts of the `MatrixClient`, whereas ideally the component does not really even need to know about the `MatrixClient`), perhaps a better choice would be to define an `interface` that has 2 required properties. This way neither `OpenIDLoader`, nor `getSFUConfigWithOpenID()` require to mention and import a `MatrixClient`: strictly speaking we don't really want to have a subset of a `MatrixClient`, we just need some type `T`, such that satisfies a given `interface` (2 fields), that way the code remains more flexible and also less "external dependencies" 🙂 What do you think?

Reply:
I was reluctant to start duplicating properties of matrixClient, but yeah, you could also do it that way, I don't really mind particularly.

Reply:
So my main concern about passing the `MatrixClient` around is that it's rather a large and powerful object, most components don't really need to know about what `MatrixClient` is, so instead of passing a large object, the components should have simple interfaces/types/structures that they expect as an input. The code would then become less coupled - the `getSFUConfigWithOpenID()` would work fine even without a matrix client, i.e. it would work with any type `T` that can provide a device ID and an open ID token.

Such things don't look particularly bad at first sight, but I decided to mention it since it's easier to explain with simpler examples. It gets much more complicated when components take many parameters, each of which relies on some large data types even if they intend to use 2-3 properties of it, but because of the type it also forces their parent components and callers to pass the large type around. As mentioned I don't want to block the merge of this PR, so I've already approved it with the current changes, but thought that it may make sense to clarify some points for the future and upcoming PRs 🙂